practice/src/js/helpers/validateForm.js

```diff
@@ -0,0 +1,142 @@
+ * @returns {string} - An error message if the value is not an integer, otherwise an empty string
+ */
+const validateInteger = ({ key, value }) =>
+  (formError[key] = !REGEX_PATTERNS.integerRegex.test(value)
+    ? `${key} must be an integer.`
+    : '');
+
+/**
+ * Checks if the value is a number
+ * @param {Object} params - An object
+ * @param {string} params.key - The field that needs to check
+ * @param {number} params.value - The value of that field
+ * @returns {string} - An error message if the value is not a float, otherwise an empty string
+ */
+const validateFloat = ({ key, value }) =>
+  (formError[key] = !REGEX_PATTERNS.digitRegex.test(value)
+    ? `${key} must be a float.`
+    : '');
+
+/**
+ * Checks if the value is a string
+ * @param {Object} params - An object
+ * @param {string} params.key - The field that needs to check
+ * @param {string} params.value - The value of that field
+ * @returns {string} - An error message if the value is not a string, otherwise an empty string
+ */
+
+const validateString = ({ key, value }) =>
+  (formError[key] =
+    typeof value !== 'string' ? `${key} must be a string.` : '');
+
+/**
+ * Checks if the value is valid format
+ * @param {Object} params - An object
+ * @param {string} params.key - The field that needs to check
+ * @param {string} params.value - The value of that field
+ * @returns {string} - An error message if the value is not a valid image URL, otherwise an empty string
+ */
+const validateImage = ({ key, value }) =>
+  (formError[key] = !REGEX_PATTERNS.imageUrlRegex.test(value)
+    ? `Please make sure ${key} starts with 'http://' or 'https://' and ends with .jpg or .png`
+    : '');
+
+/**
+ * Checks if the value is longer than minimum length
+ * @param {Object} params - An object
+ * @param {string} params.key - The field that needs to check
+ * @param {string} params.value - The value of that field
+ * @param {number} [params.min=3] - The minimum length required (default is 2)
+ * @returns {string} - An error message if the length requirement is not met, otherwise an empty string
+ */
+const validateLength = ({ key, value, min = 2, max = 50 }) =>
+  (formError[key] =
+    value.trim().length < min || value.trim().length > max
+      ? `${key} must have between ${min} and ${max} characters.`
+      : '');
+
+/**
+ * Checks if the value is a positive number
+ * @param {Object} params - An object
+ * @param {string} params.key - The field that needs to check
+ * @param {number} params.value - The value of that field
+ * @returns {string} - An error message if the value is not a positive number, otherwise an empty string
+ */
+const validatePositive = ({ key, value }) =>
+  (formError[key] =
+    parseFloat(value) < 0 ? `${key} must be a positive number.` : '');
+
+/**
+ * Validates the product form data
+ * @param {Object} data - The product form data
+ * @returns {Object} An object containing validation results
+ */
+function validateProductForm(data) {
```

Comment:
You should create a common validateForm that you can use for multiple forms, with validationSchema as an argument.